### DISCUSSION TITLE
collapsed group to show not only label, but icon or diagram

### DISCUSSION THREAD

**Discussion by tetemes** [jgraph/drawio#2310] [Ideas]
I would like to be able to see not only text but selected element(s) from the group when collapsing a group.

I can create a group of a box with selecting the box and clicking Arrange/Group menu then I am able to place items inside this group and collapse/expand on demand.

Currently:
Only the label of the box is shown when the box is collapsed.

Request:
I would like the collapsed group to display not only the box's text label, but also an icon with its text when the group is collapsed (auto-sizing/manual-sizing), so that I can make the group visually more understandable. Once this capability is available, I would be able to create expandable/collapsible architecture diagrams.

Nice addition:
If it was possible to have a switch to expand/collapse everything in the same level (groups could be selected to be in the same level), where I can switch between context, container, component and code level of details from the C4 architecture diagram model, that would complete the feature nicely. This was I would not draw so many separate diagrams with different detail levels, but would be able to maintain a single diagram and it would contain details of multiple levels.

Thank you

**Comment by kassemdarwish** [comment]
I just thought about the exact same thing while experimenting with the C4 model earlier today.

**Comment by Philzen** [comment]
Our current workaround is to create multiple pages for context, containers and components and then simply place a link on the Containers or Software Systems boxes. Works like a charm with the diagram viewer, except for navigating back (so we're simply added backlinks manually to the diagram titles, to be able to go up in the hierarchy again).

**Comment by allmhuran** [comment]
I believe my stackoverflow question (below) is a request for essentially the same feature. I have included sample images for the behaviour I would like to see.

https://stackoverflow.com/q/75668931/7165279
